src/textual/widgets/_text_area.py

```diff
@@ -684,26 +678,22 @@ def _watch_show_vertical_scrollbar(self) -> None:
         self._rewrap_and_refresh_virtual_size()
         self.scroll_cursor_visible()
 
-    def _watch_theme(self, theme: str | None) -> None:
+    def _watch_theme(self, theme: str) -> None:
         """We set the styles on this widget when the theme changes, to ensure that
-        if padding is applied, the colours match."""
+        if padding is applied, the colors match."""
         self._set_theme(theme)
 
-    def _app_dark_toggled(self) -> None:
+    def _app_dark_toggled(self):
         self._set_theme(self._theme.name)
 
-    def _set_theme(self, theme: str | None) -> None:
+    def _set_theme(self, theme: str) -> None:
         theme_object: TextAreaTheme | None
-        if theme is None:
-            # If the theme is None, use the default.
-            theme_object = TextAreaTheme.default()
-        else:
-            # If the user supplied a string theme name, find it and apply it.
-            try:
-                theme_object = self._themes[theme]
-            except KeyError:
-                theme_object = TextAreaTheme.get_builtin_theme(theme)
 
+        # If the user supplied a string theme name, find it and apply it.
+        try:
+            theme_object = self._themes[theme]
+        except KeyError:
+            theme_object = TextAreaTheme.get_builtin_theme(theme)
```

Comment:
Since you nested the `raise` below in a `try: ... except: ...` block, should you do `raise ThemeDoesNotExist(...) from None`?